pkg/transforms/argoapplication.go

```diff
@@ -73,16 +98,61 @@ func ArgoApplicationResourceBuilder(a *ArgoApplication) *ArgoApplicationResource
 	node.Properties["chart"] = a.Spec.Source.Chart
 	node.Properties["repoURL"] = a.Spec.Source.RepoURL
 	node.Properties["targetRevision"] = a.Spec.Source.TargetRevision
+
 	if a.Spec.Source.TargetRevision == "" {
 		node.Properties["targetRevision"] = "HEAD"
 	}
 
 	// Status properties
-	node.Properties["status"] = a.Status.Health.Status
+	node.Properties["healthStatus"] = a.Status.Health.Status
+	node.Properties["syncStatus"] = a.Status.Sync.Status
+
+	// resource list properties
+	resourcesByte, err := yaml.Marshal(a.Status.Resources)
+	if err != nil {
+		glog.Errorf("Marshaling ArgoCD application %v/%v resource list failed: %v", a.Namespace, a.Name, err)
+	} else {
+		node.Properties["resources"] = string(resourcesByte)
+	}
+
+	// conditions properties
+	truncatedConditions := []ApplicationCondition{}
+
+	for _, condition := range a.Status.Conditions {
+		newCondition := ApplicationCondition{
+			Type:    condition.Type,
+			Message: TruncateText(condition.Message, 512),
+		}
+
+		truncatedConditions = append(truncatedConditions, newCondition)
+	}
+
+	truncatedConditionsByte, err := yaml.Marshal(truncatedConditions)
+	if err != nil {
+		glog.Errorf("Marshaling ArgoCD application %v/%v condition list failed: %v", a.Namespace, a.Name, err)
+	} else {
+		node.Properties["conditions"] = string(truncatedConditionsByte)
+	}
 
 	return &ArgoApplicationResource{node: node}
 }
 
+func TruncateText(text string, width int) string {
+	if width < 0 {
+		return ""
+	}
+
+	res := []rune(text)
+
+	if len(res) > width {
+		res = res[:width]
+
+		return string(res) + "..."
+	}
+
+	return string(res)
```

Comment:
Will this work for you?
```suggestion
		if len(text) > width {
		return text[:width] + "..."
	}
	return text
```

Reply:
ok.